layers/+lang/c-c++/funcs.el

```diff
@@ -225,28 +225,27 @@ and the arguments for flyckeck-clang based on a project-specific text file."
   (spacemacs//c-c++-lsp-set-config (intern (concat "c-c++-lsp-" suffix)) nil (concat "-" suffix)))
 ;; -- END helper functions for common configuration of cquery and ccls backends
 
-
-(defun spacemacs//c-c++-lsp-enable ()
-  "Enable the LSP backend specified by the `c-c++-backend' configuration variable."
-    (progn (condition-case nil
-             (spacemacs//c-c++-lsp-call-function "lsp-" "-enable")
-             (user-error nil))))
-
 (defun spacemacs//c-c++-lsp-config ()
   "Configure the LSP backend specified by the `c-c++-backend' configuration variable."
     (progn
-      (spacemacs//c-c++-lsp-setup-company)
       (spacemacs//c-c++-lsp-define-extensions)
       (spacemacs//c-c++-lsp-wrap-functions)
       (setq-default flycheck-disabled-checkers '(c/c++-clang c/c++-gcc))
 
+      (dolist (param '("executable" "extra-init-params" "project-whitelist" "project-blacklist" "sem-highlight-method"))
+        (spacemacs//c-c++-lsp-apply-config param))
+
       (if (eq c-c++-lsp-cache-dir nil)
         (progn
           (setq c-c++-lsp-cache-dir (file-truename(concat "~/.emacs.d/.cache/" (symbol-name c-c++-backend))))
           (message (concat "c-c++: No c-c++-lsp-cache-dir specified: defaulting to " c-c++-lsp-cache-dir))))
 
-      (dolist (param '("executable" "extra-init-params" "cache-dir" "project-whitelist" "project-blacklist" "sem-highlight-method"))
-        (spacemacs//c-c++-lsp-apply-config param))
+      (ecase c-c++-backend
+        ('lsp-cquery (setq cquery-cache-dir c-c++-lsp-cache-dir))
+        ('lsp-ccls (setq ccls-initialization-options
+                     (if ccls-initialization-options
+                       (cons ccls-initialization-options `(:cacheDirectory ,c-c++-lsp-cache-dir))
```

Comment:
I'm not sure `cons` is a good choice here. If ccls-initialization-options is a list like `(:a "1" :b "2")` then it would create something like `((:a "1 :b "2") :cacheDirectory "<path>")` which isn't correct

```suggestion
                       (append ccls-initialization-options `(:cacheDirectory ,c-c++-lsp-cache-dir))
```

Reply:
Corrected -- I need to learn a bit of lisp properly at some point :)